### PR TITLE
Create DeleteWorkspaceAvatar

### PR DESCRIPTION
**Pull request by Gonals**
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/219963

### Tests/ QA Steps
1. In Newdot, create a workspace if you don't have one and open the General settings:
Screen Shot 2022-08-17 at 4 08 10 PM

2. If you don't have an avatar, add one:
Screen Shot 2022-08-17 at 4 08 19 PM

3. Once the avatar is added, remove it:
Screen Shot 2022-08-17 at 4 18 41 PM

4. Confirm it immediately goes back to the default avatar.
5. Add an avatar again.
6. Go offline
7. Remove the avatar. It should immediately go back to the default avatar, but it should display semitransparent:
Screen Shot 2022-08-17 at 4 38 06 PM

8. Go back online. The default avatar should go back to fully opaque.
9. Add an avatar again. Do not leave the General settings.
10. In a different tab, open your account in OldDot and go to Policies->group and delete the workspace:
Screen Shot 2022-08-17 at 4 18 29 PM

11. Go back to newdot and remove the avatar. After a bit, an error should show up:
Screen Shot 2022-08-17 at 4 01 09 PM

12. Click the X. The error should go away.

- [x] Verify that no errors appear in the JS console

### CODE REVIEW

**Comment by Gonals** [comment]
OFF WIP!

File: src/libs/actions/Policy.js [hunk 1]
```diff
@@ -378,6 +380,66 @@ function updateLocalPolicyValues(policyID, values) {
...
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    avatarURL: null,
+                },
+                errorFields: {
+                    avatarURL: {
+                        [DateUtils.getMicroseconds()]: 'Sorry, there was an unexpected problem deleting your workspace avatar.',
```
Comment (neil-marcellini) [inline src/libs/actions/Policy.js:423]: I recently learned from [this Slack post](https://expensify.slack.com/archives/C03TQ48KC/p1660838478754859) that we are supposed to use `Localize.translateLocal` when setting errors on the front end.

File: src/libs/actions/Policy.js [hunk 2]
```diff
@@ -378,6 +380,66 @@ function updateLocalPolicyValues(policyID, values) {
     Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, values);
 }
 
+function deleteWorkspaceAvatar(policyID) {
```
Comment (neil-marcellini) [inline src/libs/actions/Policy.js:383]: Please add a JSDoc documenting the param type.

File: src/libs/actions/Policy.js [hunk 3]
```diff
@@ -378,6 +380,66 @@ function updateLocalPolicyValues(policyID, values) {
...
+                        [DateUtils.getMicroseconds()]: 'Sorry, there was an unexpected problem deleting your workspace avatar.',
+                    },
+                },
+            },
+        },
+    ];
+    API.write('DeleteWorkspaceAvatar', {policyID}, {optimisticData, successData, failureData});
+}
+
+function clearAvatarErrors(policyID) {
```
Comment (neil-marcellini) [inline src/libs/actions/Policy.js:432]: JSDoc please.

**Review by neil-marcellini — CHANGES_REQUESTED** [review]
Looking good. I have a couple of comments.

I will be adding the RBR to get to the general settings page in [this PR](https://github.com/Expensify/App/pull/10400), so only adding the error under the avatar here is fine. Also, it's very unlikely that we will see errors with this flow.

**Comment by Gonals** [comment]
Comments addressed!!

File: src/languages/en.js [hunk 4]
```diff
@@ -264,6 +264,7 @@ export default {
         removePhoto: 'Remove photo',
         editImage: 'Edit photo',
         imageUploadFailed: 'Image upload failed',
+        deleteWorkspaceError: 'Sorry, there was an unexpected problem deleting your workspace avatar.',
```
Comment (neil-marcellini) [inline src/languages/en.js:267]: NAB: I was told to get a copy check for a similar error message. [Here](https://github.com/Expensify/Expensify/issues/219957#issuecomment-1222177053) is where that conversation is happening.

**Review by neil-marcellini — APPROVED** [review]
Good to go! One NAB comment. I would update the title to [HOLD Web-E 34598] so it's clear what we're holding on.

**Review by Luke9389 — APPROVED** [review]

**Comment by Gonals** [comment]
OFF HOLD!

**Comment by neil-marcellini** [comment]
It should still be on HOLD until the Web PR is deployed, otherwise it might fail on staging and create deploy blockers.

**Comment by Gonals** [comment]
> It should still be on HOLD until the Web PR is deployed, otherwise it might fail on staging and create deploy blockers.

Really? I thought they could move in parallel! My bad!

**Comment by Gonals** [comment]
Web PR is on Staging. This is OFF HOLD!
@MonilBhavsar, go for it!

**Comment by neil-marcellini** [comment]
> Web PR is on Staging. This is OFF HOLD!

I appreciate the urgency, but App staging calls the production Web api, so we need to wait until it's on prod 😞

**Comment by Gonals** [comment]
> > Web PR is on Staging. This is OFF HOLD!
> 
> I appreciate the urgency, but App staging calls the production Web api, so we need to wait until it's on prod 😞

Oh, damn :(

**Review by MonilBhavsar — APPROVED** [review]
Looks good and works well!

**Comment by neil-marcellini** [comment]
[Web](https://github.com/Expensify/Expensify/issues/219963) was just deployed to production so we can merge!

**Comment by Julesssss** [comment]
Testing well in staging

Screenshot 2022-08-26 at 12 26 44

**Pull request merged**